Dockerfile

```diff
@@ -1,13 +1,23 @@
-FROM node:12.18.1-alpine
-RUN apk add --update\
- python \
- python3 \
- build-base \
- zeromq-dev \
- && rm -rf /var/cache/apk/*
+FROM ubuntu:20.04
+
+ENV NODE_VERSION=12.18.1
+ENV NVM_DIR=/usr/local/.nvm
+
+RUN mkdir -p ${NVM_DIR}
+RUN apt-get update && apt-get install -y curl
+
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash
+RUN . "$NVM_DIR/nvm.sh" && nvm install ${NODE_VERSION}
+RUN . "$NVM_DIR/nvm.sh" && nvm use v${NODE_VERSION}
+RUN . "$NVM_DIR/nvm.sh" && nvm alias default v${NODE_VERSION}
+ENV PATH="${NVM_DIR}/versions/node/v${NODE_VERSION}/bin/:${PATH}"
+RUN node --version
+RUN npm --version
+
+RUN npm install -g yarn
```

Comment:
each of these RUNs creates a separate docker layer, which will lead to _some_ bloat. nothing is particularly wrong with it, but it's probably not what you want.

@gimre-xymcity or @Wayonb can probably give more details about this.

Reply:
Good catch about nvm.

We can just install the node distro without nvm, that's useful for dev but not production.

https://github.com/nodesource/distributions/blob/master/README.md

Reply:
The code you shared would download the latest `12.x` version not `12.18.1` so the `nvm` choice was intentional, to be able to specify the exact version `12.18.1` (since didn't feel safe to change the `node` version at this stage). I also didn't like the idea of having `nvm` in production, but during my research saw that it was the common way to handle it smoothly since it abstracts the installing process. If we want to have the latest `12.x` version, I'll be happy to change it.

Reply:
Ofc, fix version means explicit version upgrades. A bit more overhead to maintain but safer.

Reply:
Yet another comment, travis uses the latest 12 every build. 

https://github.com/symbol/catapult-rest/blob/dev/.travis.yml#L4

So we are unit testing ~100% coverage with the latest node 12 version.

Reply:
I'm keen to upgrade it to the latest 12.x tbh, just checked the [nodejs 12.x change log](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V12.md) between 12.18.1 and 12.2.6(latest) seen a lot of security fixes which is good. I assume it's safe to upgrade but we'll need proper testing and some soak time in testnet before mainnet release. 
What do you think @Jaguar0625 and @Wayonb ?
1. Upgrade it to the nodejs latest version(12.x) in the docker file (which will have the latest 12.x in each build)
2. keep nodejs version fixed at 12.18.1 (as it is)

Reply:
Can't really help with that, but would _probably_ use `fixed` version